crates/rspack_plugin_esm_library/src/link.rs

```diff
@@ -471,7 +480,12 @@ impl EsmLibraryPlugin {
       let exports_type = module.build_meta().exports_type;
       let default_object = module.build_meta().default_object;
 
-      let info = &mut concate_modules_map[id];
+      /*
+       * Safety:
+       * Scope hoisted module canonly exist in one chunk
+       * One chunk deconflict symbols won't access module in other chunk
```

Comment:
If we want to use unsafe, the correct way is use `IdentifierIndexMap<UnsafeCell<ModuleInfo>>`.

A better way is to prove it to type system. We can lock the item (like `IdentifierIndexMap<AtomicRefCell<T>>`) or preemptively take `&mut T` to prove they don't intersect.